server/src/main/java/org/opensearch/cluster/routing/allocation/allocator/BalancedShardsAllocator.java

```diff
@@ -284,6 +307,20 @@ private void setThreshold(float threshold) {
         this.threshold = threshold;
     }
 
+    private void setAllocatorTimeout(TimeValue allocatorTimeout) {
+        this.allocatorTimeout = allocatorTimeout;
+    }
+
+    protected boolean allocatorTimedOut(long currentTime) {
```

Comment:
Lets take a TimeValue as the parameter type to ensure consumers don't use the wrong time units

Reply:
Didn't get this. How can we represent current absolute time in TimeValue?

Reply:
Why do we need the consumer to pass the current time? Anyways they are passing System.nanoTime so do we really need to ask for time?

Reply:
Good point. Refactored it